### DISCUSSION TITLE
Dump with only read access

### DISCUSSION THREAD

**Discussion by phbernard** [planetscale/discussion#574] [Q&A] [answered]
TL; DR The only way I know to dump the database is to use the pscale CLI tool. It requires a PlanetScale user, who has too many rights for this task. The classic `mysqldump` is not an option, as the database it too huge.

I want to dump my database on a daily basis, automatically, for backup. I want these backups to be outside of PlanetScale for additional security.

I used to use `mysqldump`, with dedicated, read-only credentials. It ceased to work a few days ago to do the database size.

Now the only way I find to dump the database is the CLI, with something like `pscale database dump mydb main`. As far as I know, I need a PlanetScale user for this, who is either a Admin or a Member. Even the Member has too many access rights for this task.

Is there a way to dump the database while giving the process in charge on a read access to a single branch of the database?

**Comment by orware** [comment, marked as answer]
Hello @phbernard,

For what you described, using our service token approach with limited permissions is probably what you would require for a use case like this:
https://planetscale.com/docs/concepts/service-tokens#add-database-permissions

Hopefully this leads you in the right direction and it should be good for automation type usage as well!

**Comment by phbernard** [comment]
Great! That's right, this is the way to go.

It seems that the database dump (`pscale db dump <db> <branch>`) requires at least the `connect_branch` permission. This lets the service token connect to the database with read/write permission. It would be more secure to grant the service token a read access only. Is that possible?

**Reply by mscoutermarsh** [reply]
A service token with read only `connect_branch` access isn't possible at the moment.

The options are `connect_branch` and `connect_production_branch`. Both of those include read/write.
